### ISSUE TITLE
In 2 pass mode split video by GOP and encode in parallel

### ISSUE THREAD

**Issue by tleydxdy** [xiph/rav1e#1048]
could be when scene changes completely.

**Comment by tleydxdy** [comment]
just a short test using ffmpeg to split up the video, the speed is at/above x265 just with this.

**Comment by tdaede** [comment]
Yeah, I'm not sure if it's better to do this in an external tool or build it into rav1e. We do have a feature that dumps keyframe locations to a file: https://github.com/xiph/rav1e/commit/50269cb681031d6bdd5e7cfa9e55538e288f58ea

**Comment by tleydxdy** [comment]
I think it would be better if rav1e can do this, at least until proper multi-thread, sticking ivfs back together with ffmpeg kind of worked, but when I tries to get sample frames from it ffmpeg through a lot of error.

**Comment by lu-zero** [comment]
I'll try to get something I was working on out soon.

**Comment by tleydxdy** [comment]
is that segment mutithreading?

**Comment by shssoichiro** [comment]
I have a WIP tool to do this https://github.com/rust-av/rav1e-by-gop but it currently doesn't support two-pass. It probably could by using rav1e's first pass to generate the keyframe list instead of av-scenechange. At that point I wonder if it would be better to build it in to rav1e.

**Comment by tleydxdy** [comment]
looking at https://github.com/xiph/rav1e/commit/50269cb681031d6bdd5e7cfa9e55538e288f58ea it seems to use first-pass data seems quite easy
